python/ray/serve/_private/replica.py

```diff
@@ -359,6 +359,15 @@ def _wrap_user_method_call(self, request_metadata: RequestMetadata):
             yield
         except asyncio.CancelledError as e:
             user_exception = e
+
+            # Recursively cancel child requests
+            requests_pending_assignment = (
+                ray.serve.context._get_requests_pending_assignment(
+                    request_metadata.request_id
+                )
+            )
+            for t in list(requests_pending_assignment.values()):
```

Comment:
Nit: Maybe spell out `task` so it's more understanding what this is iterating over 🙃 

Also, we don't need to cast it to a list before iterate through it right?